crates/python_extractor/tests/data/nonascii.py

```diff
@@ -0,0 +1,4 @@
+# coding: iso-8859-5
```

Comment:
could we not detect this and use the correct character set to decode?

https://docs.python.org/3/reference/lexical_analysis.html#encoding-declarations


> If a comment in the first or second line of the Python script matches the regular expression coding[=:]\s*([-\w.]+),  this comment is processed as an encoding declaration; the first group of this expression names the encoding of the source code file.


Seems easy enough to parse with a regex. Take the first two lines and look for `#.*coding[=:]\s*([-\w.]+)` and then look up the character set by string. If it isn't found, then use utf-lossy like you have here.